packages/serverless-workflow-language-service/src/channel/SwfLanguageServiceCodeCompletion.ts

```diff
@@ -129,6 +138,190 @@ function getStateNameCompletion(
+      remoteList.push(dataInputSchemaPath);
+    } else {
+      relativeList.push(dataInputSchemaPath);
+    }
+  }
+  if (remoteList.length > 0 || relativeList.length > 0) {
+    const schemaData = await Promise.all([
+      ...(await args.jqCompletions.remote.getJqAutocompleteProperties({
+        textDocument: args.document,
+        schemaPaths: remoteList ?? [],
+      })),
+      ...(await args.jqCompletions.relative.getJqAutocompleteProperties({
+        textDocument: args.document,
+        schemaPaths: relativeList ?? [],
+      })),
+    ]);
+    if (schemaData.length === 0) {
+      return Promise.resolve([]);
+    }
+    return Promise.resolve(
+      schemaData
+        .filter((prop: Record<string, string>) => {
+          if (args.wordToSearch === ".") {
+            return true;
+          } else {
+            return Object.keys(prop)[0].startsWith(args.wordToSearch);
+          }
+        })
+        .map((parsedProp: Record<string, string>) => {
+          return createCompletionItem({
+            ...args,
+            completion: Object.keys(parsedProp)[0],
+            kind: CompletionItemKind.Value,
+            label: Object.keys(parsedProp)[0],
+            detail: Object.values(parsedProp)[0],
+            filterText: args.wordToSearch === "." ? Object.keys(parsedProp)[0] : args.wordToSearch,
+            overwriteRange: Range.create(
+              Position.create(
+                args.cursorPosition.line,
+                args.wordToSearch === "."
+                  ? args.cursorPosition.character
+                  : args.cursorPosition.character - args.wordToSearch.length
+              ),
+              Position.create(args.cursorPosition.line, args.cursorPosition.character)
+            ),
+          });
+        })
+    );
+  }
+  return [];
+}
+/**
+ * get reusable functions defined in the functions array of the swf file.
+ */
+function getReusableFunctionCompletion(
+  args: SwfLanguageServiceCodeCompletionFunctionsArgs & { wordToSearch: string }
+): CompletionItem[] {
+  const reusalbeFunctions: SwfLsNode = findNodeAtLocation(args.rootNode, ["functions"])!;
+  const functionNamesArray: string[] = [];
+  if (reusalbeFunctions.type === "array") {
+    reusalbeFunctions.children?.forEach((func) => {
+      if (findNodeAtLocation(func, ["type"])?.value === "expression") {
+        const functionName = findNodeAtLocation(func, ["name"])?.value;
+        functionNamesArray.push(functionName);
+      }
+    });
+    return functionNamesArray
+      .filter((name: string) => {
+        if (args.wordToSearch === "") {
+          return true;
+        }
+        return name.startsWith(args.wordToSearch);
+      })
+      .map((filteredName: string) => {
+        return createCompletionItem({
+          ...args,
+          completion: filteredName,
+          kind: CompletionItemKind.Function,
+          label: filteredName,
+          filterText: args.wordToSearch.length ? args.wordToSearch : filteredName,
```

Comment:
Together with my previous comment

```suggestion
          filterText: isWordToSearchEmpty ? args.wordToSearch : filteredName,
```